### PR TITLE
Geyser fixes

### PR DESCRIPTION
**Pull request by Obsidianninja11**
- Added a max distance to geyser box
- Added option to only show geyser box while holding lava rod
- holdingLavaRod and holdingWaterRod in fishingAPI

https://discord.com/channels/997079228510117908/1211104046845530122

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/fishing/FishingAPI.kt [hunk 1]
```diff
@@ -52,8 +59,19 @@ object FishingAPI {
...
+
+
+    @SubscribeEvent
+    fun onRepoReload(event: RepositoryReloadEvent) {
+        val data = event.getConstant<ItemsJson>("Items")
+        lavaRods = data.lava_fishing_rods ?: error("§clava_fishing_rods is missing from repo.")
+        waterRods = data.water_fishing_rods ?: error("§cwater_fishing_rods is missing from repo.")
     }
 
+
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/fishing/FishingAPI.kt]: remove empty line

File: src/main/java/at/hannibal2/skyhanni/features/fishing/trophy/GeyserFishing.kt [hunk 2]
```diff
@@ -69,7 +71,11 @@ class GeyserFishing {
 
     @SubscribeEvent
     fun onRenderWorld(event: LorenzRenderWorldEvent) {
-        if (!config.drawBox || !IslandType.CRIMSON_ISLE.isInIsland()) return
+        if (!config.drawBox) return
+        if (!IslandType.CRIMSON_ISLE.isInIsland()) return
+        if (config.onlyWithRod && !FishingAPI.holdingLavaRod) return
+        if (geyser?.distanceToPlayerIgnoreY()!! > 96) return
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/fishing/trophy/GeyserFishing.kt]: use `val geyser = geyser ?: return` instead of `!!`

**Review by hannibal002 — CHANGES_REQUESTED** [review]
works fine in game

File: src/main/java/at/hannibal2/skyhanni/features/fishing/trophy/GeyserFishing.kt (hunk 2 above)
Comment (Obsidianninja11) [reply src/main/java/at/hannibal2/skyhanni/features/fishing/trophy/GeyserFishing.kt]: so shadow geyser? (I think that's what it's called)

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**